### PR TITLE
feat: categories redirect action on click developed with styling

### PR DESCRIPTION
**Pull request by dev-enforced**
In this Pull request the reviewer would be able to view the files related to an implementation of a feature that involves categories provided on MI Stream.
So on the home page as the reviewer scrolls to the bottom he would find certain categories clicking on which would redirect him to the video listing page where he/she will be shown the videos shown based on the category clicked earlier.
On the video listing page there are certain chips which helps the reviewer get access to videos of a certain category.

Here are the links: https://mi-stream-git-categories-redirect-dev-enforced.vercel.app/
https://mi-stream-git-categories-redirect-dev-enforced.vercel.app/explore

### CODE REVIEW

File: src/components/Categories/Categories.jsx [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
+import { updateCategorySelected } from "reduxFiles";
 import styles from "./Categories.module.css";
 const Categories = ({ categoriesGiven }) => {
+  const dispatch=useDispatch();
+  const navigate=useNavigate();
   return (
-    <div className={`${styles.categoriesContainer}`}>
+    <div className={`${styles.categoriesContainer}`} >
       {categoriesGiven?.map((everyCategory) => {
         const {_id,categoryName,imgUrl,description}=everyCategory;
```
Comment (harshalini) [inline src/components/Categories/Categories.jsx:12]: I like that you destructured it

File: src/hooks/useOnClickOutside.js [hunk 2]
```diff
@@ -11,10 +11,12 @@ const useOnClickOutside = (eventHandler, referenceGiven) => {
     };
     document.addEventListener("touchstart", listenerHandler);
     document.addEventListener("mousedown", listenerHandler);
+    document.addEventListener("mouseleave",listenerHandler);
 
     return () => {
       document.removeEventListener("touchstart", listenerHandler);
       document.removeEventListener("mousedown", listenerHandler);
+      document.addEventListener("mouseleave", listenerHandler);
```
Comment (harshalini) [inline src/hooks/useOnClickOutside.js:19]: This is really nice!

**Review by harshalini — COMMENTED** [review]
Really good implementation of redux, everything is written in separate files 👍

**Review by kavyakunder — APPROVED** [review]
The code looks clean and is easily understandable. Looks good to merge!

File: src/hooks/useOnClickOutside.js [hunk 3]
```diff
@@ -11,10 +11,12 @@ const useOnClickOutside = (eventHandler, referenceGiven) => {
     };
```
Comment (Namrata-J) [inline src/hooks/useOnClickOutside.js:11]: Good to see that you have used the use effect here to do things in the dom.

File: src/utilities/VideoShufflers/videoShufflerFunction.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  for (let i = shuffledVideoList.length - 1; i > 0; i--) {
+    let j = Math.floor(Math.random() * (i + 1));
+    [shuffledVideoList[i], shuffledVideoList[j]] = [
+      shuffledVideoList[j],
+      shuffledVideoList[i],
+    ];
+  }
+  return shuffledVideoList;
+};
+
```
Comment (Namrata-J) [inline src/utilities/VideoShufflers/videoShufflerFunction.js:12]: the code here is so well structured!!

**Review by Namrata-J — APPROVED** [review]

**Pull request merged**